src/scenic/core/geometry.py

```diff
@@ -10,322 +10,329 @@
-			raise RuntimeError(f'unknown kind of shapely geometry {polygon}')
+    if polygon.is_empty:
+        return
+    if isinstance(polygon, (shapely.geometry.MultiPolygon,
+                            shapely.geometry.MultiLineString,
+                            shapely.geometry.MultiPoint,
+                            shapely.geometry.collection.GeometryCollection)):
+        polygons = polygon.geoms
+    else:
+        polygons = [polygon]
+    for polygon in polygons:
+        if isinstance(polygon, shapely.geometry.Polygon):
+            yield polygon.exterior
+            for ring in polygon.interiors:
+                yield ring
+        elif isinstance(polygon, (shapely.geometry.LineString, shapely.geometry.LinearRing,
+                                  shapely.geometry.Point)):
+            yield polygon
+        else:
+            raise RuntimeError(f'unknown kind of shapely geometry {polygon}')
 
 def plotPolygon(polygon, plt, style='r-', **kwargs):
-	for chain in allChains(polygon):
-		x, y = chain.xy
-		plt.plot(x, y, style, **kwargs)
+    for chain in allChains(polygon):
+        x, y = chain.xy
+        plt.plot(x, y, style, **kwargs)
 
 class _RotatedRectangle:
-	"""mixin providing collision detection for rectangular objects and regions"""
-	def containsPoint(self, point):
-		pt = shapely.geometry.Point(point)
-		return self.polygon.intersects(pt)
-
-	def intersects(self, rect):
-		return self.polygon.intersects(rect.polygon)
-
-	@cached_property
-	def polygon(self):
-		position, heading, hw, hl = self.position, self.heading, self.hw, self.hl
-		if any(needsSampling(c) or needsLazyEvaluation(c)
-		       for c in (position, heading, hw, hl)):
-			return None		# can only convert fixed Regions to Polygons
-		corners = _RotatedRectangle.makeCorners(position.x, position.y, heading, hw, hl)
-		return shapely.geometry.Polygon(corners)
-
-	@staticmethod
-	def makeCorners(px, py, heading, hw, hl):
-		s, c = sin(heading), cos(heading)
-		s_hw, c_hw = s*hw, c*hw
-		s_hl, c_hl = s*hl, c*hl
-		corners = (
-			(px + c_hw - s_hl, py + s_hw + c_hl),
-			(px - c_hw - s_hl, py - s_hw + c_hl),
-			(px - c_hw + s_hl, py - s_hw - c_hl),
-			(px + c_hw + s_hl, py + s_hw - c_hl)
-		)
-		return corners
+    """mixin providing collision detection for rectangular objects and regions"""
+    def containsPoint(self, point):
+        pt = shapely.geometry.Point(point)
+        return self.polygon.intersects(pt)
+
+    def intersects(self, rect):
+        return self.polygon.intersects(rect.polygon)
+
+    @cached_property
+    def polygon(self):
+        position, heading, hw, hl = self.position, self.heading, self.hw, self.hl
+        if any(isLazy(c) for c in (position, heading, hw, hl)):
+            return None     # can only convert fixed Regions to Polygons
+        corners = _RotatedRectangle.makeCorners(position.x, position.y, heading, hw, hl)
+        return shapely.geometry.Polygon(corners)
+
+    @staticmethod
+    def polygonFromObj(obj):
```

Comment:
This isn't used anywhere and probably has been superseded by `Object._boundingPolygon` in any case.